File: 458. Poor Pigs/solution.js
/**
 * @param {number} buckets
 * @param {number} minutesToDie
 * @param {number} minutesToTest
 * @return {number}
 */
var poorPigs = function(buckets, minutesToDie, minutesToTest) {
  let pigs = 0;
  while (Math.pow(minutesToTest / minutesToDie + 1, pigs) < buckets) pigs++;
  return pigs;
};

File: 458. Poor Pigs/solution.go
import (
	"math"
)

func poorPigs(buckets int, minutesToDie int, minutesToTest int) int {
	if buckets <= 0 {
		return 0
	}

	if minutesToTest/minutesToDie < 1 {
		return 0 // no answer
	}

	if minutesToTest/minutesToDie == 1 {
		return int(math.Ceil(math.Log2(float64(buckets))))
	}

	base := minutesToTest/minutesToDie + 1
	n := math.Log(float64(buckets)) / math.Log(float64(base))
	n = math.Ceil(n)
	return int(n)
}